services/QuillLMS/app/services/classlink_integration/district_importer.rb

```diff
@@ -2,24 +2,25 @@
 
 module ClasslinkIntegration
   class DistrictImporter < ApplicationService
-    attr_reader :district_oneroster_id, :district_bearer_token, :district_id, :tenant_id
+    attr_reader :district_oneroster_id,
+      :district_bearer_token,
+      :district_id,
+      :tenant_id,
+      :premium_school_external_ids
 
     def initialize(district_oneroster_id:, district_bearer_token:, district_id:, tenant_id:)
       @district_oneroster_id = district_oneroster_id
       @district_bearer_token = district_bearer_token
       @district_id = district_id
       @tenant_id = tenant_id
+      @premium_school_external_ids = fetch_premium_school_external_ids
     end
 
-    def run
-      validate_relationships!
-      persist_records
-    end
+    def run = persist_records
```

Comment:
I would either get rid of periste_records passthrough or make it a private method.